specification/network/resource-manager/Microsoft.Network/stable/2021-08-01/expressRoutePort.json

```diff
@@ -867,6 +1113,13 @@
           "type": "string",
           "description": "Date of the physical port allocation to be used in Letter of Authorization."
         },
+        "authorizations": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ExpressRoutePortAuthorization"
+          },
+          "description": "The list of authorizations."
+        },
         "links": {
```

Comment:
If the authorizations are modelled as a child of the expressRoutePorts then why do also need them in the properties ? Is the data being duplicated here ? It is recommended to just leave the data in the child and client make a call to get the list collection instead of doing it this way.

Reply:
The child resource as a property help us to add and remove it by doing a PUT on the parent. Data is not duplicated but rather a reference of the child is saved in the parent. NRP allows to serialize/deserialize the child resource at the time of parent GET\PUT. A similar property is “links” here which is also a child resource but also added as a property of parent. 

Also a similar property called authorizations is already exists for [Express Route Circuit resource](https://github.com/Azure/azure-rest-api-specs/blob/network-2021-08-01/specification/network/resource-manager/Microsoft.Network/stable/2021-08-01/expressRouteCircuit.json) and we want the experience of Express Route Ports to be consistent with Circuit. Let me know if you have more questions.

Reply:
@raosuhas please let me know if there are more concerns/questions.

Reply:
> Have left one comment. Please let me know if that field was added by mistake. Rest looks good

@raosuhas please take a look. have responded to your comment

Reply:
This is an ARM anti-pattern, ref: https://armwiki.azurewebsites.net/api_contracts/guidelines/resourcegraph.html#arg001-do-not-model-embedded-resource-or-reference-child-resource-qualifications-on-parent-resources-such-as-counts.

There are a bunch of reasons modeling this way is not a good idea. The RPC will be adopting this language (or similar):
1. RPs MUST NOT model a subresource as BOTH nested and an embedded array for the following reasons
  - Change on the nested resource type needs notification on 2 types
  - Change on the parent resource type will need to conditionally emit notifications based on what changed
  - Plus the reasons below
2. Modelling a subresource ONLY as an embedded array SHOULD be avoided for the following reasons
  - When there are thousands of instances possible for the subresource, the overall payload size can go larger than the overall paylof limit for control plane operations. Is this 8MB?
  - Pagination cannot be implemented
  - RBAC cannot be provided for the subresource separate from the parent
  - Computing changes can get complex

When the children are an unordered set, individual items should be mutated via individual CRUD without doing a full GET of the parent.

I understand that this pattern is in heavy use by NRP APIs. ARM would like to move away from this pattern.

Reply:
resolved